client/app/visualizations/pivot/Editor.jsx

```diff
@@ -1,23 +1,57 @@
 import { merge } from 'lodash';
-import React from 'react';
+import React, { useState } from 'react';
 import Switch from 'antd/lib/switch';
 import { EditorPropTypes } from '@/visualizations';
 
 export default function Editor({ options, onOptionsChange }) {
+  const [hideRowTotals, setHideRowTotals] = useState(
+    options.rendererOptions &&
+      options.rendererOptions.table &&
+      options.rendererOptions.table.rowTotals === false,
+  );
```

Comment:
You can add defaults to visualization's default options and then omit all that tests: https://github.com/getredash/redash/blob/9a4433bf685eec1f3fb24639ee11ab3e9fd8b115/client/app/visualizations/pivot/index.js#L13-L17

Also, no need to use state here: just use `const hideRowTotals = options.rendererOptions.table.rowTotals === false;` or pass `checked={options.rendererOptions.table.rowTotals === false}` where needed (the same for second option).

Reply:
> You can add defaults to visualization's default options and then omit all that tests:

Two questions:

1. Will this work with existing visualizations?
2. `rendererOptions` might have other properties set by PivotTable.js so I don't want to override it. Will it get merged or overriden?

> Also, no need to use state here: just use...

This is what I did at first, but then it wouldn't update then I change it...

Reply:
1. yes
2. it's merged, so everything should be fine.

> it wouldn't update then I change it...

That's strange :thinking: Try to add defaults to `DEFAULT_OPTIONS` and check will it work or not (it may be related to `_.merge`). If it will not work - leave everything as is, I'll check check it when migrating this to React.

Reply:
Done. https://github.com/getredash/redash/pull/3943/commits/e313dcbab1b4a8673174c257ea23ac1bad470603